Fresh work on nowrap cells

andyjeffries opened the pull request:
Based almost entirely on:

https://github.com/prawnpdf/prawn-table/pull/50

Implements the wrapping, adds no more failures (than the current master fails on 3.0.4 anyway), but adds new tests and documentation.

andyjeffries committed:
Fresh work on nowrap cells

Based almost entirely on:

https://github.com/prawnpdf/prawn-table/pull/50

Implements the wrapping, adds no more failures (than the current master fails on 3.0.4 anyway), but adds new tests and documentation.

andyjeffries commented:
I took the work on #50, made it not fail any more tests than before, added new tests and docs. All seems "fine", except there's an additional border now, that I don't know how to get rid of!

Any ideas @gettalong ? Thanks.

![Screenshot 2022-09-09 at 13 26 14](https://user-images.githubusercontent.com/22904/189349563-8ef8c616-fc60-4faf-b1f3-76b9dafc3ca2.png)

andyjeffries commented:
I've just done another example in there to debug, and if I force the left cell to have a newline, the right cell is effectively only half-height, but doesn't get the weird border mid-cell.

![Screenshot 2022-09-09 at 14 54 50](https://user-images.githubusercontent.com/22904/189366668-66de4e31-cf21-403e-8546-14538b3e9d92.png)

gettalong commented:
@andyjeffries I'm not very acquainted with the prawn-table code but will have a look if I find some time.

andyjeffries committed:
Adjust documentation for nowrap to not draw an individual border

andyjeffries commented:
Thanks anyway @gettalong , I found it though. I was calling `cell` in the manual which immediately draws it to the page as a standalone object (with its own height, that of the content), then passed it to the table (which drew it at the same height as its neighbouring cell).

This is now ready for review and merging (ignoring the test that failed before this one)

andyjeffries committed:
Add whitespace option to nowrap

andyjeffries commented:
Just added `whitespace` as an option to nowrap. I found in my app that if I don't use `nowrap` it can wrap mid-word awkwardly in a table, but if i I use the new `nowrap` it's too long for the rest of the row to fit nicely. That is not what I intended, so now I've added another option to handle that. The manual should hopefully make it clear.

![Screenshot 2022-09-12 at 11 33 24](https://user-images.githubusercontent.com/22904/189633017-205ef546-6e3e-48d5-a71b-23e382805b11.png)

andyjeffries committed:
Don't try to compare nil, if the content is accidentally nil

andyjeffries committed:
Further fix for trying to compare with nil

andyjeffries commented:
@gettalong this is ready for review, but the request is blocked (I've been using it in production for six months now). Could you approve/review it?

gettalong commented:
@andyjeffries I'm sorry but I currently don't have the time for that.
